src/main/java/cinemas/repositories/impl/UserRepositoryImpl.java

```diff
@@ -2,11 +2,29 @@
 
 import cinemas.models.User;
 import cinemas.repositories.UserRepository;
+import jakarta.persistence.NoResultException;
+import jakarta.persistence.TypedQuery;
 import org.springframework.stereotype.Repository;
 
+import java.util.List;
+
+
 @Repository("userRepository")
 public class UserRepositoryImpl extends BaseRepositoryImpl<User, Integer> implements UserRepository {
     public UserRepositoryImpl() {
         super(User.class);
     }
+
+    @Override
+    public User findByEmail(String email) {
+        String hql = "FROM User u WHERE u.email = :email";
+        TypedQuery<User> query = entityManager.createQuery(hql, User.class);
+        query.setParameter("email", email);
+
+        try {
+            return query.getSingleResult();
+        } catch (NoResultException e) {
+            return null;
+        }
```

Comment:
=)) try catch đặt ở service ấy em chư sko phải repostory nhé 
```suggestion
            return query.getSingleResult();
```